plugins/engines/ospray/OSPRayScene.cpp

```diff
@@ -138,13 +155,28 @@ void OSPRayScene::commit()
                 transformation * instance.getTransformation();
 
             if (modelDescriptor->getBoundingBox() && instance.getBoundingBox())
+            {
+                // scale and move the unit-sized bounding box geometry to the
+                // model size/scale first, then apply the instance transform
+                const auto& modelBounds =
+                    modelDescriptor->getModel().getBounds();
+                Transformation modelTransform;
+                modelTransform.setTranslation(modelBounds.getCenter() /
+                                                  modelBounds.getSize() -
+                                              Vector3f(0.5f));
+                modelTransform.setScale(modelBounds.getSize());
```

Comment:
I don't quite understand what you're doing here.
- The geometry of the unit-sized bb, is centered at 0, 0, 0, or 0.5, 0.5, 0.5?
- modelBound.getSize() is returning a 3d vector or a float?
As I see it, the bounding box gets distorted in funny ways is the model is much larger in one dimension that the other two. Was it already the case? If not I-d prefer another solution that doesn't make bounding boxes look deformed.

Reply:
There is no deformation, be assured.